### PR TITLE
new tests for core module

### PR DESCRIPTION
**Pull request by roomanidzee**

### CODE REVIEW

File: dsl/src/test/scala/ru/itclover/tsp/dsl/v2/FunctionRegistryTest.scala [hunk 1]
```diff
@@ -1,97 +1,183 @@
 package ru.itclover.tsp.dsl.v2
+
 import org.scalactic.{Equality, TolerantNumerics}
 import org.scalatest.{FlatSpec, Matchers}
 import org.scalatest.prop.PropertyChecks
 import ru.itclover.tsp.core.Result
 
 class FunctionRegistryTest extends FlatSpec with Matchers with PropertyChecks {
-  val funReg = DefaultFunctionRegistry
+  val funReg: DefaultFunctionRegistry.type = DefaultFunctionRegistry
```
Comment (hyst329) [inline dsl/src/test/scala/ru/itclover/tsp/dsl/v2/FunctionRegistryTest.scala]: As far as I know, one can write just `val funReg: FunctionRegistry = DefaultFunctionRegistry`

Comment (roomanidzee) [reply dsl/src/test/scala/ru/itclover/tsp/dsl/v2/FunctionRegistryTest.scala]: no, in that case compiler can't find case class for function registry

**Comment by tampler** [comment]
Works for me, but some tests fail for DSL and Core. @fabura @hyst329 pls take a look

**Comment by roomanidzee** [comment]
fixed some typos in tests, but something happened in dsl module.

@hyst329, look please at it.

**Pull request merged**